lib/services/api/authentication_service.rb

```diff
@@ -0,0 +1,10 @@
+module Api
+  class AuthenticationService
+    def self.create_authentication(manager_resource, attrs)
```

Comment:
Minor, but I feel like this should be named `create_authenticaton_queue` or `create_authenticaton_task` or something.  Then the caller knows they are getting back a task, and it's clear that it's not a synchronous call.